### ISSUE TITLE
Extremely large number of columns result in errors (a metadata detection issue)

### ISSUE THREAD

**Issue by MoHeydarian** [galaxyproject/galaxy#3796]
Datasets with an extremely large number of columns are not processed as they should be with simple Galaxy tools (like 'Filter' and 'Compare two datasets'). 

"Metadata detection doesn't do well with huge numbers of columns" per @natefoo. 

With the addition of the Transpose tool to Galaxy, users may encounter this metadata detection issue. Some tools ('Compare two datasets') will return green output datasets that are actually empty when encountering files with 'too many columns'.

**Comment by dannon** [comment]
I can look up the changesets in the morning, but I wanted to drop the reference here while I'm thinking of it.  I'm pretty sure @jmchilton has fixed related issues in the past and we might be able to generalize the approaches a bit -- swapping from reading the whole file to NN bytes, etc.

**Comment by mvdbeek** [comment]
Yeah, that's an annoying one, and it's not only with huge numbers of columns. Large GFF and BED (3 or more GB) files get read in completely. Metadata setting on these can easily take 20 minutes. That is probably related.

**Comment by jmchilton** [comment]
I try to modify datatype sniffers so they don't consume unbounded memory on upload. This problem is I guess that datasets may have unbounded metadata size - which we know we don't want to store in the database (and we explicitly prohibit). I would say the path forward here is:

- Tweak metadata collection on tabular files to collect info on only the first N columns (configurable)
- Add a new boolean metadata field to tabular datasets ``extra_columns``.
- In the tool form allow text input of column index if ``extra_columns`` is set.
- Fix the particular tools to do their own column parsing if ``extra_columns`` is set - we need to rely a lot less on Galaxy's limited metadata handling in these circumstances.

Existing regressions in the tool form such as https://github.com/galaxyproject/galaxy/issues/3758 may need to be addressed also.

**Comment by natefoo** [comment]
Previously there was a handler memory consumption problem with loading the set_meta output when there were large numbers of columns, I am not sure if this is still the case. IIRC this was the very problem that @bgruening encountered while we were in Australia, although that may have been a huge number of temp files issue.

**Comment by bgruening** [comment]
In my case it was a huge number of temp files generated with `parallelization` mode. Which causes problems with every restart as they are not tracked in the DB.
